### ISSUE TITLE
Failed test with Cap'n Proto 0.7.0 tools

### ISSUE THREAD

**Issue by sannyschulz** [capnproto/go-capnproto2#149]
go test -v zombiezen.com/go/capnproto2 

failed for tests:

TestMarshalShouldMatchTextWhenDecoded
TestMarshalPackedShouldMatchTextWhenDecoded

with tool version Cap'n Proto 0.7.0 (using executables from https://capnproto.org/capnproto-c++-win32-0.7.0.zip )

--- FAIL: TestMarshalShouldMatchTextWhenDecoded (0.34s)
    integration_test.go:739: zdataFilledMessage(20): decoded to:
        "(zdata = (data = \"\\000\\001\\002\\003\\004\\005\\006\\a\\b\\t\\n\\v\\f\\r\\016\\017\\020\\021\\022\\023\"))\n"; want:
        "(zdata = (data = \"\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\a\\b\\t\\n\\v\\f\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\"))\n"
--- FAIL: TestMarshalPackedShouldMatchTextWhenDecoded (0.34s)
    integration_test.go:762: zdataFilledMessage(20): decoded to:
        "(zdata = (data = \"\\000\\001\\002\\003\\004\\005\\006\\a\\b\\t\\n\\v\\f\\r\\016\\017\\020\\021\\022\\023\"))\n"; want:
        "(zdata = (data = \"\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\a\\b\\t\\n\\v\\f\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\"))\n"
		

For tool version Cap'n Proto 0.5.0 these tests still work.

**Comment by sannyschulz** [comment]
Happens with marshalTest:

	{
		name: "zdataFilledMessage(20)",
		msg:  zdataFilledMessage(t, 20),
		typ:  "Z",
		text: `(zdata = (data = "\x00\x01\x02\x03\x04\x05\x06\a\b\t\n\v\f\r\x0e\x0f\x10\x11\x12\x13"))` + "\n",
		data: []byte{
			0, 0, 0, 0, 9, 0, 0, 0,
			0, 0, 0, 0, 3, 0, 1, 0,
			28, 0, 0, 0, 0, 0, 0, 0,
			0, 0, 0, 0, 0, 0, 0, 0,
			0, 0, 0, 0, 0, 0, 0, 0,
			0, 0, 0, 0, 0, 0, 1, 0,
			1, 0, 0, 0, 162, 0, 0, 0,
			0, 1, 2, 3, 4, 5, 6, 7,
			8, 9, 10, 11, 12, 13, 14, 15,
			16, 17, 18, 19, 0, 0, 0, 0,
		},
	},


It happens also if **test.msg.Marshal()** is replaced by **tool.Encode(...)**

**Comment by lthibault** [comment]
I'm getting this behavior for `v2.17.4` as well.

``` 
$ go test zombiezen.com/go/capnproto2/...
--- FAIL: TestMarshalPackedShouldMatchTextWhenDecoded (0.70s)
    integration_test.go:762: zdataFilledMessage(20): decoded to:
        "(zdata = (data = \"\\000\\001\\002\\003\\004\\005\\006\\a\\b\\t\\n\\v\\f\\r\\016\\017\\020\\021\\022\\023\"))\n"; want:
        "(zdata = (data = \"\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\a\\b\\t\\n\\v\\f\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\"))\n"
--- FAIL: TestMarshalShouldMatchTextWhenDecoded (0.74s)
    integration_test.go:739: zdataFilledMessage(20): decoded to:
        "(zdata = (data = \"\\000\\001\\002\\003\\004\\005\\006\\a\\b\\t\\n\\v\\f\\r\\016\\017\\020\\021\\022\\023\"))\n"; want:
        "(zdata = (data = \"\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\a\\b\\t\\n\\v\\f\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\"))\n"
FAIL
FAIL    zombiezen.com/go/capnproto2     1.114s
ok      zombiezen.com/go/capnproto2/capnpc-go   1.187s
ok      zombiezen.com/go/capnproto2/encoding/text       0.257s
?       zombiezen.com/go/capnproto2/internal/aircraftlib        [no test files]
?       zombiezen.com/go/capnproto2/internal/capnptool  [no test files]
ok      zombiezen.com/go/capnproto2/internal/demo       0.533s
?       zombiezen.com/go/capnproto2/internal/demo/books [no test files]
?       zombiezen.com/go/capnproto2/internal/demo/hashes        [no test files]
ok      zombiezen.com/go/capnproto2/internal/fulfiller  0.195s
?       zombiezen.com/go/capnproto2/internal/nodemap    [no test files]
ok      zombiezen.com/go/capnproto2/internal/packed     6.755s
ok      zombiezen.com/go/capnproto2/internal/queue      0.213s
?       zombiezen.com/go/capnproto2/internal/schema     [no test files]
?       zombiezen.com/go/capnproto2/internal/strquote   [no test files]
ok      zombiezen.com/go/capnproto2/pogs        0.452s
ok      zombiezen.com/go/capnproto2/rpc 0.240s
?       zombiezen.com/go/capnproto2/rpc/internal/logtransport   [no test files]
?       zombiezen.com/go/capnproto2/rpc/internal/logutil        [no test files]
?       zombiezen.com/go/capnproto2/rpc/internal/pipetransport  [no test files]
ok      zombiezen.com/go/capnproto2/rpc/internal/refcount       0.296s
?       zombiezen.com/go/capnproto2/rpc/internal/testcapnp      [no test files]
ok      zombiezen.com/go/capnproto2/schemas     0.187s
ok      zombiezen.com/go/capnproto2/server      0.190s
?       zombiezen.com/go/capnproto2/std/capnp/cxx       [no test files]
?       zombiezen.com/go/capnproto2/std/capnp/json      [no test files]
?       zombiezen.com/go/capnproto2/std/capnp/persistent        [no test files]
?       zombiezen.com/go/capnproto2/std/capnp/rpc       [no test files]
?       zombiezen.com/go/capnproto2/std/capnp/rpctwoparty       [no test files]
?       zombiezen.com/go/capnproto2/std/capnp/schema    [no test files]```

**Comment by lthibault** [comment]
@sannyschulz Have you found a fix for this?  Does it seem to be causing problems?

**Comment by titanxxh** [comment]
> @zombiezen I'm getting this behavior for `v2.17.4` as well.
> 
> ```
> $ go test zombiezen.com/go/capnproto2/...
> --- FAIL: TestMarshalPackedShouldMatchTextWhenDecoded (0.70s)
>     integration_test.go:762: zdataFilledMessage(20): decoded to:
>         "(zdata = (data = \"\\000\\001\\002\\003\\004\\005\\006\\a\\b\\t\\n\\v\\f\\r\\016\\017\\020\\021\\022\\023\"))\n"; want:
>         "(zdata = (data = \"\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\a\\b\\t\\n\\v\\f\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\"))\n"
> --- FAIL: TestMarshalShouldMatchTextWhenDecoded (0.74s)
>     integration_test.go:739: zdataFilledMessage(20): decoded to:
>         "(zdata = (data = \"\\000\\001\\002\\003\\004\\005\\006\\a\\b\\t\\n\\v\\f\\r\\016\\017\\020\\021\\022\\023\"))\n"; want:
>         "(zdata = (data = \"\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\a\\b\\t\\n\\v\\f\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\"))\n"
> FAIL
> FAIL    zombiezen.com/go/capnproto2     1.114s
> ok      zombiezen.com/go/capnproto2/capnpc-go   1.187s
> ok      zombiezen.com/go/capnproto2/encoding/text       0.257s
> ?       zombiezen.com/go/capnproto2/internal/aircraftlib        [no test files]
> ?       zombiezen.com/go/capnproto2/internal/capnptool  [no test files]
> ok      zombiezen.com/go/capnproto2/internal/demo       0.533s
> ?       zombiezen.com/go/capnproto2/internal/demo/books [no test files]
> ?       zombiezen.com/go/capnproto2/internal/demo/hashes        [no test files]
> ok      zombiezen.com/go/capnproto2/internal/fulfiller  0.195s
> ?       zombiezen.com/go/capnproto2/internal/nodemap    [no test files]
> ok      zombiezen.com/go/capnproto2/internal/packed     6.755s
> ok      zombiezen.com/go/capnproto2/internal/queue      0.213s
> ?       zombiezen.com/go/capnproto2/internal/schema     [no test files]
> ?       zombiezen.com/go/capnproto2/internal/strquote   [no test files]
> ok      zombiezen.com/go/capnproto2/pogs        0.452s
> ok      zombiezen.com/go/capnproto2/rpc 0.240s
> ?       zombiezen.com/go/capnproto2/rpc/internal/logtransport   [no test files]
> ?       zombiezen.com/go/capnproto2/rpc/internal/logutil        [no test files]
> ?       zombiezen.com/go/capnproto2/rpc/internal/pipetransport  [no test files]
> ok      zombiezen.com/go/capnproto2/rpc/internal/refcount       0.296s
> ?       zombiezen.com/go/capnproto2/rpc/internal/testcapnp      [no test files]
> ok      zombiezen.com/go/capnproto2/schemas     0.187s
> ok      zombiezen.com/go/capnproto2/server      0.190s
> ?       zombiezen.com/go/capnproto2/std/capnp/cxx       [no test files]
> ?       zombiezen.com/go/capnproto2/std/capnp/json      [no test files]
> ?       zombiezen.com/go/capnproto2/std/capnp/persistent        [no test files]
> ?       zombiezen.com/go/capnproto2/std/capnp/rpc       [no test files]
> ?       zombiezen.com/go/capnproto2/std/capnp/rpctwoparty       [no test files]
> ?       zombiezen.com/go/capnproto2/std/capnp/schema    [no test files]```
> ```

same issue with you, wondering why?

**Comment by zombiezen** [comment]
It's not actually harmful: the `capnp` tool must have changed its text output format (which isn't guaranteed not to change) for escape sequences. Such a brittle test should likely be removed or made less brittle.

PRs welcome.

**Comment by zenhack** [comment]
@lthibault fixed this in #169, closing.
